lib/rules/container-name-pattern/__tests__/index.mjs

```diff
@@ -0,0 +1,119 @@
+import rule from '../index.mjs';
+
+const { messages, ruleName } = rule;
+
+testRule({
+	ruleName,
+	config: ['foo-.+'],
+
+	accept: [
+		{ code: 'a { color: red; }' },
+		{ code: '@container NONE {}' },
+		{ code: '@container foo-bar {}' },
+		{ code: ' @CONTAINER foo-bar   {}' },
+		{ code: ' @container foo-bar {}' },
+		{ code: '@container foo-bar {\n}' },
+		{ code: '@container foo-bar (width > 400px) {}' },
+		{ code: '@container foo-bar () {}' },
+		{ code: '@container foo-bar style(--cards small) {}' },
+		{ code: '@container foo-bar scroll-state(stuck: top) {}' },
+		{ code: '@container foo-bar (inline-size > 30em), foo-baz style(--large: true) {}' },
+		{ code: '@container foo-bar (inline-size > 30em) and foo-baz style(--large: true) {}' },
+		{
+			code: '@container () {}',
+			description: 'nameless container condition',
+		},
+		{
+			code: 'a { container-name: foo-bar; }',
+		},
+		{
+			code: 'a { container: foo-bar / inline-size; }',
+		},
+	],
+
+	reject: [
+		{
+			code: '@container foo {}',
+			message: messages.expected('foo', 'foo-.+'),
+			line: 1,
+			column: 12,
+			endLine: 1,
+			endColumn: 15,
+		},
+		{
+			code: '@container bar {}',
+			message: messages.expected('bar', 'foo-.+'),
+			line: 1,
+			column: 12,
+			endLine: 1,
+			endColumn: 15,
+		},
+		{
+			code: '@container FOO-bar {}',
+			message: messages.expected('FOO-bar', 'foo-.+'),
+			line: 1,
+			column: 12,
+			endLine: 1,
+			endColumn: 19,
+		},
+		{
+			code: '@container foo-bar (), FOO-bar () {}',
+			message: messages.expected('FOO-bar', 'foo-.+'),
+			line: 1,
+			column: 24,
+			endLine: 1,
+			endColumn: 31,
+		},
+		{
+			code: 'a { container-name: FOO-bar; }',
+			message: messages.expected('FOO-bar', 'foo-.+'),
+			line: 1,
+			column: 21,
+			endLine: 1,
+			endColumn: 28,
+		},
+		{
+			code: '@container bar (inline-size > 30em), baz style(--large: true) {}',
+			warnings: [
+				{
+					message: messages.expected('bar', 'foo-.+'),
+					line: 1,
+					column: 12,
+					endLine: 1,
+					endColumn: 15,
+				},
+				{
+					message: messages.expected('baz', 'foo-.+'),
+					line: 1,
+					column: 38,
+					endLine: 1,
+					endColumn: 41,
+				},
+			],
+		},
+	],
+});
+
+testRule({
+	ruleName,
+	config: [/^foo-bar$/],
+
+	accept: [
+		{
+			code: '@container foo-bar {}',
+			description: 'Accepts pattern in RegExp notation',
+		},
+	],
+
+	reject: [
+		{
+			code: '@container foo-baz {}',
+			description: 'Accepts pattern in RegExp notation',
```

Comment:
question

Both the reject and the accept share the same description, is that on purpose?

Reply:
Let's remove the 2nd one.